src/rng/test/kat_cpp.cpp

```diff
@@ -98,7 +98,7 @@ void dev_execute_tests(kat_instance *tests, unsigned ntests) {
   unsigned i;
   for (i = 0; i < ntests; ++i) {
     kat_instance *ti = &tests[i];
-    genmap_t::iterator p = genmap.find(make_pair(ti->method, ti->nrounds));
+    auto p = genmap.find(make_pair(ti->method, ti->nrounds));
```

Comment:
I still don't like how clang tidy chooses to go to `auto` here. I thought it was only supposed to do this when they type was obvious from the RHS?

Reply:
After double checking, I confirmed that `clang-tidy` doesn't like this.  Here's the rule:
```
.../src/rng/test/kat_cpp.cpp:101:5: warning: use auto when declaring iterators [modernize-use-auto]
    genmap_t::iterator p = genmap.find(make_pair(ti->method, ti->nrounds));
    ^~~~~~~~~~~~~~~~~~
    auto
```

I'm sympathetic to your concern, but in this case I think `auto` actually makes sense. The actual data type is defined 5 lines above and the scope of use in limited to the next 3 lines of code.  That actual data type isn't really descriptive in this context.  I think I will leave it as `auto`.